### ISSUE TITLE
Support Dart syntax higlighting

### ISSUE THREAD

**Issue by JaredEzz** [continuedev/continue#1478]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

Other languages have syntax highlighting in the Continue response that make reading and using the response much more helpful.

Dart (and I'm sure other languages) do not have this support

![image](https://github.com/continuedev/continue/assets/13698465/b5fb6dc8-99f4-4325-9116-229f4a0c12ff)

### Solution

Add support for Dart and other languages for syntax highlighting in the response

**Comment by sestinj** [comment]
@JaredEzz We use react-markdown to render code blocks and other markdown with comes with built-in highlight-js. Highlight-js has a default set of included languages which apparently must not include Dart.

One possible solution would be to default to a similar language that is included, by making an override map here: https://github.com/continuedev/continue/blob/preview/gui/src/components/markdown/StyledMarkdownPreview.tsx#L141-L145

**Comment by JaredEzz** [comment]
@sestinj thanks. Dart is included in the highlight.js supported languages here.. why would it require an override map? https://github.com/highlightjs/highlight.js/blob/main/SUPPORTED_LANGUAGES.md

**Comment by JaredEzz** [comment]
If continue is using the default minified web build, it could require extra work. I'll start there

> Note: The languages available will depend on how you've built or are included the library in your app. For example our default minified web build includes only ~40 popular languages. See [Getting the Library](https://github.com/highlightjs/highlight.js#getting-the-library) and [Importing the Library](https://github.com/highlightjs/highlight.js#importing-the-library) in the README for examples of how to load additional/specific languages.

**Comment by JaredEzz** [comment]
`continuedev/continue/gui/src/components/markdown/StyledCode.tsx` seems to be the culprit, since there's no direct highlight.js imports but instead using [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) which supports languages like Dart. Not sure what the issue is here, could use some help.

**Comment by JaredEzz** [comment]
@sestinj

**Comment by JaredEzz** [comment]
@conorhastings you might have some input here too, to see if the usage is incorrect

**Comment by JaredEzz** [comment]
Agreed. I don't have the time at the moment to work on this fix, but I still would love to see it as continue is still my chosen LLM tool during development.
